### PR TITLE
Add contributing doc

### PR DESCRIPTION
**Pull request by bruth**
Draft of an initial repo-specific contributing guide (this links to the general NATS one).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+- `main` - tracks changes for the current minor series, e.g. `2.9.x`
+- `dev` - tracks changes for the first release of next minor series, e.g. `2.10.0`
+
+Once `2.10.0` is released, `main` will then track patch releases for the `2.10.x` series and `dev` will be bumped to track the `2.11.0` release.
+
+_Note: there may be various other branches present in the repo, owned by core maintainers having branching permission on the repo, however only `main` and `dev` track changes that will result in releases._
+
+## Contributing code
+
+Before writing any code, the branch the changes will be propsoed on needs to be determined.
```
Comment (MauriceVanVeen) [inline CONTRIBUTING.md:27]:
```suggestion
Before writing any code, the branch the changes will be proposed on needs to be determined.
```

**Comment by bruth** [comment]
@wallyqs I should have added `[ci skip]`, but do mind kicking the failed test in CI?

**Comment by derekcollison** [comment]
@bruth where are we on this one?

**Comment by bruth** [comment]
Superseded by #4619

**Pull request closed without merging**